### PR TITLE
feat: Desktop Referral widget

### PR DESCRIPTION
**Pull request by Tulun**
Screen Shot 2022-06-28 at 2 28 13 PM

*Overview**

Styles for Referral Widget.
Note: Only for Desktop, responsive CSS coming.

**Acceptance Criteria**
Should match design

### CODE REVIEW

File: src/views/Rewards/comp/RewardReferral/RewardReferral.tsx [hunk 1]
```diff
@@ -1,45 +1,105 @@
...
+              <LearnMoreText>
+                Learn more <ArrowUpRight />
+              </LearnMoreText>
+            </ConnectRow>
+          )}
+        </ReferralLinkBlock>
+        <ReferralTierBlock>
+          <TierSmHeader>Current referral tier</TierSmHeader>
+          <TierHeader>Bronze</TierHeader>
+          <Stepper currentStep={2} numSteps={5} />
```
Comment (daywiss) [inline src/views/Rewards/comp/RewardReferral/RewardReferral.tsx:77]: is it right these values are going to eventually be calculated by some api call or client?

File: src/components/Stepper/useStepper.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect, useCallback } from "react";
+import { StyledStepperItem } from "./Stepper.styles";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCheck } from "@fortawesome/free-solid-svg-icons";
+
+export default function useStepper(currentStep: number, numSteps: number) {
```
Comment (daywiss) [inline src/components/Stepper/useStepper.tsx:6]: would be great to get some comments on what this hook does and how it works, the logic is hard to follow

**Review by daywiss — COMMENTED** [review]
just want to get some clarity on what the stepper function is doing since theres no way to really test this.  it seems though like a good candidate for a unit test

File: src/views/Rewards/comp/RewardReferral/RewardReferral.tsx (hunk 1 above)
Comment (Tulun) [reply src/views/Rewards/comp/RewardReferral/RewardReferral.tsx:77]: Yup. This is just a stub.

NumSteps should be static (it's just the amount in the stepper itself), but current step would be calculated based on what tier they are on.

File: src/components/Stepper/useStepper.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const [stepItems, setStepItems] = useState<JSX.Element[]>([]);
+  const createStepsItems = useCallback(() => {
+    const items: JSX.Element[] = [];
+    const midPoint = Math.round(numSteps / 2);
+    for (let i = 0; i < numSteps; i++) {
+      const completed = i + 1 < currentStep;
```
[the remark this answers is not in the document]
Comment (Tulun) [reply src/components/Stepper/useStepper.tsx:12]: Sure, I can readjust this to start at i = 1.

**Comment by nicholaspai** [comment]
Some QA suggestions

- Boxes are too close to side edge of screen in non-full screen mode 
Screen Shot 2022-06-29 at 15 10 10
- Clicking copy doesn't just copy the URL as expected, it copies this: "Across blurb www.across.to/referrer=0x...."

Otherwise LGTM!

File: src/components/Stepper/useStepper.tsx (hunk 2 above)
Comment (daywiss) [reply src/components/Stepper/useStepper.tsx:10]: i see so this kind of does css dynamically based on the number of steps

**Review by daywiss — APPROVED** [review]

File: src/components/Stepper/useStepper.tsx (hunk 2 above)
Comment (amateima) [reply src/components/Stepper/useStepper.tsx:10]: I see this hook and the associated CSS too complicated for what we want to achieve. I think we can have the same result with some relatively simple CSS as [showcased here](https://jsfiddle.net/dyjt05gf/). Sorry if I didn't fully understand it and my example doesn't cover your needs.

Screenshot 2022-06-29 at 22 36 00

File: src/views/Rewards/comp/ConnectTableOverlay/ConnectTableOverlay.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import styled from "@emotion/styled";
+import { SecondaryButtonWithoutShadow } from "components/Buttons";
+import { COLORS } from "utils";
+
+export const Overlay = styled.div`
+  position: absolute;
+  width: 100%;
+  max-width: 1406px;
+  margin-top: -200px;
+  left: 11%;
```
Comment (amateima) [inline src/views/Rewards/comp/ConnectTableOverlay/ConnectTableOverlay.styles.tsx:10]: I don't see why this overlay should be relative to the whole < Wrapper />. This overlay and the Referrals table should be contained inside a separate container that will also have `display: relative`. This way you won't need to define `left`, `max-width` and `height` attributes. Also, please place CSS attributes in the order of their importance. `display` and `height` should be above `text-align` and `background-color`.

An example of weird behaviour caused by the `left` usage

Screenshot 2022-06-29 at 22 40 10

**Review by amateima — CHANGES_REQUESTED** [review]

File: src/components/Stepper/useStepper.tsx (hunk 2 above)
Comment (Tulun) [reply src/components/Stepper/useStepper.tsx:10]: Nice. I think it is too simple but I might be able to adapt that and avoid the use of pseudoelements.

**Review by amateima — APPROVED** [review]
